packages/big-design/src/components/AccordionGroup/AccordionPanel/AccordionPanel.tsx

```diff
@@ -0,0 +1,55 @@
+import { Text } from '@bigcommerce/big-design';
+import { ExpandMoreIcon } from '@bigcommerce/big-design-icons';
+import React, { HTMLAttributes, memo } from 'react';
+
+import { useUniqueId } from '../../../hooks';
+import { Box, BoxProps } from '../../Box/Box';
+import { useAccordion } from '../hooks/useAccordion';
+
+import { StyledAccordionButton, StyledAccordionContent } from './styled';
+
+export interface AccordionPanelProps extends HTMLAttributes<HTMLDivElement> {
+  children?: React.ReactNode;
+  defaultExpanded?: boolean;
+  header: string;
+  iconLeft?: React.ReactNode;
+  isExpanded: boolean;
+  onClick?: () => void;
```

Comment:
I think we could delete `onClick` ?

Reply:
And `isExpanded`

Reply:
Do we want to remove `isExpanded`? I thought this was needed if we wanted this to be uncontrolled / controlled. Although, I'm not 100% clear on that.

Reply:
Handling the state inside the component makes it a controlled component, which you are doing with this hook: https://github.com/bigcommerce/big-design/pull/933/files#diff-5fc0e597eae1361cf0fba2493f91154fba01f35b851baace00572488ec7ec010R22

If we want consumers to control the state, we'd elevate that outside of the component and let consumers pass it in.

Reply:
I don't think this is entirely accurate -- this component is purposefully meant to be a controlled component, hence the ability for consumers to provide `isExpanded` and `onClick`, and we are definitely elevating these as far as I can see (through the `panel` items).

We should, however, export the `useAccordion` hook (I now realize its not being exported, maybe this is the source of the disconnect?), so consumers can opt for controlled or uncontrolled accordions.

Reply:
good call about the hook 👀 , didn't remember about that lol . If that's the case `onClick` and `isExpanded` will be necessary

Reply:
Sorry for the confusion 🤦‍♀️. I moved `useAccordion` outside of the component.

Reply:
The problem I see if that those props are not even being used, so we either need to use them or lose them.

Reply:
We _could_ drop `onClick`, but `isExpanded` is literally a required prop, it will get used. Not sure I follow

Reply:
What I mean is when you pull down the component, the passed in `isExpanded` and `onClick` are technically not being used/consumed within the component. `onClick` only works because we are spreading it into `<Box {...props}>` and the handler is being called when you expand/collapse the accordion. The passed `props.isExpanded` though is not being consumed anywhere, the state is being respected only through the state returned from `useAccordion`.

Reply:
Was chatting with Chance this morning in the office, and cleared up some of the misunderstandings in this thread. The issues, as they stand right now, are:

- The component is not honoring the provided `isExpanded` / `onClick` props. To correctly visualize this, we can think of a consumer that programatically mutates this prop. Since it's not going through `onClick`, it will not work
- Because of the above, the `useAccordion` hook kinda operates on the item level, but the exported element is the group level. This also means we only need to export 1 component (the group one)
- The Hooks folder is not being exported, so we need to move the `useAccordion` hook to the component folder and probably export it from there
- Some of the names could be tweaked to make more sense, stylistic changes, etc

We can sync offline and discuss these issues, IMO we'll need to address them before we move forward